frontend/src/components/ThemedButtons/LoadingButton.tsx

```diff
@@ -1,27 +1,33 @@
 import { ThreeDots } from 'react-loader-spinner';
 
-import ThemedButton, { ThemedButtonProps } from './ThemedButton';
+import ThemedButton from './ThemedButton';
 
-import './Loader.css';
+import './LoadingButton.css';
 
 function LoadingButton({
 	children,
 	isLoading = false,
-	...buttonProps
-}: ThemedButtonProps & {
+	onClick,
+}: {
+	children: React.ReactNode;
 	isLoading?: boolean;
+	onClick: () => void;
 }) {
 	return (
-		<ThemedButton
-			disabled={isLoading}
-			appearsDifferentWhenDisabled={false}
-			{...buttonProps}
-		>
-			{children}
-			{isLoading && (
-				<ThreeDots width="1.5rem" color="white" wrapperClass="loader" />
-			)}
-		</ThemedButton>
+		<div className="loading-button">
```

Comment:
not a huge fan of this single-child div. I'd much rather apply styles directly to the themed button via an extra classname

Reply:
hmm yes I would rather get rid of the div, but I wasn't sure how to override the disabled styling on the ThemedButton otherwise. Whereabout do you suggest this extra classname should be?

Reply:
add it as a prop to themedButton, and within the code for themedButton, append it to the existing classname. Like what I did to the headerButton here https://github.com/ScottLogic/prompt-injection/pull/633/commits/ebd0b042821b4d8d0e07cbaa3055ffe09f0369f0

Reply:
elegant! done now